torch/distributed/distributed_c10d.py

```diff
@@ -1297,6 +1298,65 @@ def get_node_local_rank(fallback_rank: Optional[int] = None) -> int:
     )
 
 
+def _get_backends_for_timeout_set(group: ProcessGroup) -> Set[c10d_Backend]:
```

Comment:
Curious, do we really have use cases where a single PG has multiple backends?

Reply:
when users does not pass in backend specifically, there are multiple backends. I think @H-Huang can comment more on this.

Reply:
This seems kind of overwrought given that we don't actually support setting ephemeral timeout on anything that is not NCCL lol. I guess it is symmetric with the current timeout setting logic though.

Reply:
Yeah, we can just get backend for nccl here. So we don't touch the old python code. Do we care about other backends like gloo and MPI, probably not...